sample/sample_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package sample
+
+import (
+	"math"
+	"testing"
+
+	"github.com/gonum/stat"
+	"github.com/gonum/stat/dist"
+)
+
+type lhDist interface {
+	Quantile(float64) float64
+	CDF(float64) float64
+}
+
+func TestLatinHypercube(t *testing.T) {
+	for _, nSamples := range []int{1, 2, 5, 10, 20} {
+		samples := make([]float64, nSamples)
+		for _, dist := range []lhDist{
+			dist.Uniform{Min: 0, Max: 1},
+			dist.Uniform{Min: 0, Max: 10},
+			dist.Normal{Mu: 5, Sigma: 3},
+		} {
+			LatinHypercube(samples, dist, nil)
+			for i, v := range samples {
+				p := dist.CDF(v)
+				if p < float64(i)/float64(nSamples) || p > float64(i+1)/float64(nSamples) {
+					t.Errorf("probability out of bounds")
+				}
+			}
+		}
+	}
+}
+
+func ExampleMetropolisHastings_burnin() {
+	nSamples := 1000
+	burnin := 50
+	var initial float64
+	target := dist.Weibull{K: 5, Lambda: 0.5}
+	proposal := CondNorm{Sigma: 0.2}
```

Comment:
Probably have a comment explaining what this is. Remember this is a documentation example.